definitions.yaml

```diff
@@ -136,7 +136,29 @@ definitions:
         type: array
         items:
           $ref: '#/definitions/V4NodeDefinition'
+      port_mappings:
+        type: array
+        description: |
+          Reveals the ports on the load balancer that are mapped to this cluster's ingress
+          and which protocol that port supports. Only shown and relevant on our on-prem KVM clusters.
+        items:
+          $ref: '#/definitions/V4PortMappingDefinition'
 
+  V4PortMappingDefinition:
+    type: object
+    properties:
+      ingress_port:
+        description:
+          The port on this cluster's ingress.
+        type: integer
+      load_balancer_port:
+        description: |
+          The port on the load balancer that is mapped to this cluster's ingress.
```

Comment:
It maybe helps to use the concepts of customer/company relationship and the concept of our host cluster wording. I am biased but something like this is most clear to me personally. 
 
```
The port on the customer's load balancer that is mapped to the Giant Swarm host cluster ingress-controller.
```

Reply:
hmm thx, it gets a bit wordy for me, and doesn't help me understand.

The host cluster shouldn't even have anything to do with anything here. We're looking at details of a guest cluster. 

So this is the port on the load balancer that lets you talk to the ingress of your guest cluster. whether or not the host cluster is in between there is not relevant right?

Reply:
> The host cluster shouldn't even have anything to do with anything here.

This is how it works. I just state facts and we should be as clear as possible to make customers understand what they are doing. IMO transparency is a good thing. I also only tried to help. You can gladly do what you want here. Maybe lets just go with the best we come up and iterate on it. When we notice there is a common misunderstanding over and over again we can adapt. 

> So this is the port on the load balancer that lets you talk to the ingress of your guest cluster. whether or not the host cluster is in between there is not relevant right?

It is relevant to me in order to understand how it works. When you want to hide this to the customer, fine with me. Your sentence I quoted here is quite accurate in summary I guess.

Reply:
Thanks for the help! Wasn't rejecting it out right, just trying to get closer to the real understanding, but from the outsiders perspective.

Reply:
I think there's a misunderstanding there. The host cluster has quite a lot to do with this as we are talking about onprem here.

In onprem the host cluster ingress controller forwards TCP ports to the guest cluster ingress controller. The guest cluster ingress controller needs 2 of those (http/https). Thus, needs to forward traffic on 2 ports from his (hardware) lb to host cluster IC.

The thing we call lbport is actually a port on the host cluster (behind which the host cluster ingress controller sits). So it is not the port of the customer's load balancer. On the customer's load balancer that is usually sth like IP:443 and IP:80 (where IP is the dedicated IP for a single guest cluster).

So a typical example for a guest cluster https request would look like following:

optional outside DNS -> https://ingress.clusterid.basedomain (DNS)-> IP:443 (on hw lb) -> 30021 (on host cluster) -> host cluster IC TCP service for cluster x -> 30011 (on guest cluster ) -> guest cluster IC https -> (service (only for endpoint lookup in IC -> ) pod -> container

Reply:
Maybe `This is the port on the host cluster ingress controller that needs to be set up to receive traffic for this cluster.`

Reply:
or even minus the `to be set up` it needs to receive traffic, how you solve that, is up to you, and you most probably know if you are looking for this.